rumqttc/src/v5/eventloop.rs

```diff
@@ -406,12 +407,22 @@ async fn mqtt_connect(
     // validate connack
     match network.read().await? {
         Incoming::ConnAck(connack) if connack.code == ConnectReturnCode::Success => {
-            // Override local keep_alive value if set by server.
+            // Check if it's a new session
+            if !connack.session_present {
+                // If it's a new session, clear the pendings
+                pending.clear();
+            }
+            // Override local settings if set by server.
             if let Some(props) = &connack.properties {
+                // Override local keep_alive value if set by server.
                 if let Some(keep_alive) = props.server_keep_alive {
                     options.keep_alive = Duration::from_secs(keep_alive as u64);
                 }
                 network.set_max_outgoing_size(props.max_packet_size);
+                // Override local session_expiry_interval value if set by server.
+                if (props.session_expiry_interval).is_some() {
+                    options.set_session_expiry_interval(props.session_expiry_interval);
+                }
```

Comment:
These changes should rather go into #854

Reply:
Fine, I'll move that.